src/standard/TIP-4/6.md

```diff
@@ -0,0 +1,184 @@
+
+### TIP4_6Collection.platformCode()
+```solidity
+function platformCode() external view responsible returns (TvmCell);
+```
+
+Returns the NftPlatform code containing the `collection`(`address`) salt. The NftPlatform code is non-upgradeable.
+
+### TIP4_6Collection.platformCodeInfo()
+```solidity
+function platformCodeInfo() external view responsible returns (uint256 codeHash, uint16 codeDepth);
+```
+
+Returns the hash and depth of the NftPlatform code. These can be used to calculate the NFT address in third-party contracts. The hash is taken from the NftPlatform code containing the `collection`(`address`) salt.
+
+### TIP4_6Collection.gasUpgradeValue()
+```solidity
+function gasUpgradeValue() external view responsible returns (uint128 fixedValue, uint128 dynamicValue);
+```
+
+Returns the amount of gas required to upgrade the NFT.
+
+### Events
+```solidity
+event UpgradeNftRequested(uint32 oldVersion, uint32 newVersion, address nft, address initiator);
+event NftCodeUpdated(uint32 oldVersion, uint32 newVersion, uint256 oldCodeHash, uint256 newCodeHash);
+```
+**UpgradeNftRequested**
+
+You must emit it when an update is requested from the NFT and upgrade is available (the NFT version at the time of the upgrade request is less than the current NFT version in the collection).
+
+**NftCodeUpdated**
+
+You must emit it when NFT code is updated in a collection. The function of code update is not standardized.
+
+### Mint
+
+The functions implementing minting are not standardized. The process itself differs from the one proposed in [TIP-4.1](1.md) (see Abstract).
+
+### Upgrade
+
+When NFT code is updated in a collection, the latter changes the NFT version and emits the `NftCodeUpdated` event.
+
+## Nft
+```solidity
+interface ITIP4_6Nft {
+    event NftUpgraded(uint32 oldVersion, uint32 newVersion, address initiator);
+
+    function requestUpgrade(address sendGasTo) external;
+	
+    function version() external view responsible returns (uint32);
+}
+```
+
+### TIP4_6Nft.requestUpgrade()
+```solidity
+function requestUpgrade(address sendGasTo) external;
+```
+
+- `sendGasTo (address)` - the address to which the remaining gas will be sent
+
+Calls the Сollection function, checking if there is an upgrade. If a new version is available, the Сollection upgrades the NFT by calling the appropriate function (this function is not included in the standard, implementations may vary).
+
+### TIP4_6Nft.version()
+```solidity
+function version() external view responsible returns (uint32);
+```
+
+Returns the current version of NFT.
+
+### Events
+```solidity
+event NftUpgraded(uint32 oldVersion, uint32 newVersion, address initiator);
+```
+
+**NftUpgraded**
+
+You must emit it after the NFT upgrade.
+
+### Upgrade
+
+The NFT upgrade scenario is as follows:
+
+1. The user (in the general case, the NFT manager) calls the NFT `requestUpgrade` function.
+2. The NFT requests an upgrade from the collection, passing information about itself (`id`, `version`).
```

Comment:
Which function, precisely?

Reply:
> 

It's an internal communication between the contracts in the NFT system, and it shouldn't be specified by the standard